R/utilities-figures.R

```diff
@@ -252,29 +252,26 @@ createPlotsFromExcel <- function(simulatedScenarios, observedData, projectConfig
   # read sheet "exportConfiguration"
   dfExportConfigurations <- readExcel(file.path(projectConfiguration$paramsFolder, projectConfiguration$plotsFile), sheet = "exportConfiguration") %>%
     rename(name = outputName)
-  dfDataCombined <- .validateDataCombinedFromExcel(dfDataCombined, simulatedScenarios, observedData, stopIfNotFound)
-  dataCombinedNames <- unique(dfDataCombined$DataCombinedName)
-  dfPlotConfigurations <- .validatePlotConfigurationFromExcel(dfPlotConfigurations, dataCombinedNames)
+
   dfPlotGrids <- .validatePlotGridsFromExcel(dfPlotGrids, unique(dfPlotConfigurations$plotID))
+  # Filter and validate only used plot configurations
+  dfPlotConfigurations <- dplyr::filter(dfPlotConfigurations, plotID %in% unlist(unique(dfPlotGrids$plotIDs)))
+  dfPlotConfigurations <- .validatePlotConfigurationFromExcel(dfPlotConfigurations, unique(dfDataCombined$DataCombinedName))
+  # Filter and validate only used data combined
+  dfDataCombined <- dplyr::filter(dfDataCombined, DataCombinedName %in% unique(dfPlotConfigurations$DataCombinedName))
+  dfDataCombined <- .validateDataCombinedFromExcel(dfDataCombined, simulatedScenarios, observedData, stopIfNotFound)
 
-  # Only consider plotIDs that are specified in the plot grids
-  validPlotIDs <- as.character(unique(unlist(dfPlotGrids$plotIDs)))
```

Comment:
We assume that only used `plotIDs` are left in `dfPlotConfigurations` at this point.